### PR TITLE
expression: implement vectorized evaluation for `builtinTrim1ArgSig`

### PR DESCRIPTION
**Pull request by shihongzhi**
### What problem does this PR solve? 

implement vectorized evaluation for builtinRightSig , for #12106

### What is changed and how it works?

    $go test -v -benchmem -bench=BenchmarkVectorizedBuiltinStringFunc -run=BenchmarkVectorizedBuiltinStringFunc -args "builtinTrim1ArgSig"
	goos: darwin
	goarch: amd64
	pkg: github.com/pingcap/tidb/expression
	BenchmarkVectorizedBuiltinStringFunc/builtinTrim1ArgSig-VecBuiltinFunc-4         	   18150	     68859 ns/op	   26752 B/op	     836 allocs/op
	BenchmarkVectorizedBuiltinStringFunc/builtinTrim1ArgSig-NonVecBuiltinFunc-4      	   14688	     81380 ns/op	   26752 B/op	     836 allocs/op
	PASS
	ok  	github.com/pingcap/tidb/expression	3.973s

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_string_vec_test.go [hunk 1]
```diff
@@ -34,7 +34,9 @@ var vecBuiltinStringCases = map[string][]vecExprBenchCase{
 	ast.Unhex: {
 		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETString}, geners: []dataGenerator{&randHexStrGener{10, 100}}},
 	},
-	ast.Trim:      {},
+	ast.Trim: {
+		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETString}},
```
Comment (XuHuaiyu) [inline expression/builtin_string_vec_test.go]: The `defaultGener` will never generate a string with leading or trailing space. We may need to add a new data generator or refine the `defaultGener`.

Comment (shihongzhi) [reply expression/builtin_string_vec_test.go]: I add a new data generator which called "randSpaceStrGener"

File: expression/bench_test.go [hunk 2]
```diff
@@ -373,6 +373,29 @@ func (g *randHexStrGener) gen() interface{} {
 	return string(buf)
 }
 
+type randSpaceStrGener struct {
```
Comment (SunRunAway) [inline expression/bench_test.go]: It is not a common generator, could it be put into builtin_string_vec_test.go

Comment (shihongzhi) [reply expression/bench_test.go]: okay.  I move it to builtin_string_vec_test.go

**Comment by qw4990** [comment]
@shihongzhi Please solve the conflict.

**Comment by shihongzhi** [comment]
> @shihongzhi Please solve the conflict.

resolved

**Pull request merged**